### DISCUSSION TITLE
Periodic wifi drops from cameras

### DISCUSSION THREAD

**Discussion by blfriedman** [roleoroleo/yi-hack-MStar#469] [General]
I am running yi-hack 0.4.8 on the y23 cameras and see consistent wifi drops every half hour or less.  

I do not believe it is related to wifi signal strength or the access point.  The wifi signal strength on one camera experiencing the drops is -37 dBm as reported at the router (Asus RT-AC68P Merlin fw).  The same camera reports wifi strength of 100%.

This issue is seen only for the yi cameras amongst many other devices across four different access points.

The cameras are not rebooting, only losing the wifi and then reconnecting.  

On each camera, I have disabled everything but the RTSP video server ( I have disabled ALL of the following: RTSP audio, FTP, Cloud, recording to SD card, snapshot, ONVIF, telnet, and MQTT).

The cameras are simply providing two H.264 streams (both high and low resolution) to a single Frigate server which does all of the motion and human detection and recording external to the camera.

Looking at top, the camera app "rmm" is using around 37% cpu, and the H.264 streams are using 17-20% cpu.  At the time of the drops, just before and just after, there does not seem to be any change seen in the top results w.r.t. cpu usage.  There is at least 25% idle cpu report by top, and 2MB free memory.

Where is the wireless handled? Is it in the monolithic "rmm"?  Has anyone else experienced similar wifi disconnects?

**Comment by roleoroleo** [comment]
Wifi is handled in dispatch process (if I remeber correctly) and it's not under my control.
Are you using a repeater or a mesh network or something with 2 different APs with the same SSID?

**Reply by blfriedman** [reply]
I see this issue with both 0.4.7 and 0.4.8, and I don't assume it is new.  I thought that the issue may lie internal to the Yi portion of the firmware, but I'm just wondering if others have noticed it.

I noticed it while trying to clean up my wifi issues and discovered the duration of the connections to the APs in the  wireless log page on the AP was always less than 30 minutes.

I use separate SSIDs on four different access points, so that for each camera there is only one AP to connect to.  The APs each have a wired connection back to the router, and there is no mesh network or repeaters involved.  

I do see the closed issue "Frequent connection drops #149" from last year, and I wonder if it is related to running in private mode causing a reset somewhere in the Yi portion of the firmware.  I will try running one camera connected to the cloud to see if that can clear it up.

**Reply by blfriedman** [reply]
I did try using cloud on one camera; however, I cannot configure the system to NOT record motion events to the SD card when connected to the cloud.

I would rather live with the periodic drops than have it require and record to the SD card.  I have had instability in the past when the SD card goes bad it takes down the camera, and they go bad after a while from frequent writes.

I imagine there is some kind of error recovery happening in the Yi code when the cloud connection is lost that causes it to reset the wifi.  Too bad we can't change their code.
